example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,370 @@
+
+    return result.addWithPaintOffset(
+      offset: contentOffset,
+      position: position,
+      hitTest: (BoxHitTestResult result, Offset transformed) {
+        assert(transformed == position - contentOffset);
+        return child?.hitTest(result, position: transformed) ?? false;
+      },
+    );
+  }
+
+  /// Builds the path used to paint the menu.
+  ///
+  /// The path includes a rounded rectangle and a arrow pointing to [arrowDirection], centered at [arrowCenter].
+  Path _buildBorderPath(ArrowDirection arrowDirection, double arrowCenter) {
+    final halfOfBase = arrowBaseWidth / 2;
+
+    // Adjust the rect to leave space for the arrow.
+    // During layout, we reserve space for the arrow in both x and y axis.
+    final contentRect = Rect.fromLTWH(
+      arrowLength,
+      arrowLength,
+      size.width - arrowLength * 2,
+      size.height - arrowLength * 2,
+    );
+
+    Path path = Path()..addRRect(RRect.fromRectAndRadius(contentRect, Radius.circular(borderRadius)));
+
+    // Add the arrow.
+    if (arrowDirection == ArrowDirection.left) {
+      path
+        ..moveTo(contentRect.centerLeft.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.right) {
+      path
+        ..moveTo(contentRect.centerRight.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.up) {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.topCenter.dy)
+        ..relativeLineTo(halfOfBase, -arrowLength)
+        ..relativeLineTo(halfOfBase, arrowLength);
+    } else {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.bottomCenter.dy)
+        ..relativeLineTo(halfOfBase, arrowLength)
+        ..relativeLineTo(halfOfBase, -arrowLength);
+    }
+
+    path.close();
+
+    return path;
+  }
+
+  /// Computes the direction where the arrow should point to.
+  ///
+  /// If [globalFocalPoint] is inside the [menuRect] horizontal bounds, the arrow points up or right. Otherwise,
+  /// the arrow points left or right.
+  ///
+  /// If [allowHorizontalArrow] is `false`, the arrow only points up or down.
+  ArrowDirection _computeArrowDirection(Rect menuRect, Offset globalFocalPoint) {
+    final isFocalPointInsideHorizontalBounds =
+        globalFocalPoint.dx >= menuRect.left && globalFocalPoint.dx <= menuRect.right;
+    if (isFocalPointInsideHorizontalBounds || !allowHorizontalArrow) {
+      if (globalFocalPoint.dy < menuRect.top) {
+        return ArrowDirection.up;
+      }
+      return ArrowDirection.down;
+    } else {
+      if (globalFocalPoint.dx < menuRect.left) {
+        return ArrowDirection.left;
+      }
+      return ArrowDirection.right;
+    }
+  }
+
+  /// Computes the center point of the arrow.
```

Comment:
I don't know what the "center point of the arrow" is. Is that a position on the arrow, or a position on the menu border? What kind of value should I expect to be returned?

Reply:
```
        /\
--------------------
        |
   It would be this point when the arrow points up (it's not properly aligned, though)
```

I'm not sure what would be the correct description for this.